File: frontend/src/api/bsfAria.ts
/**
 * BSF ARIA API (Module 16). ARIA explains, summarises, recommends and answers —
 * advisory only, read-only, never mutating a plan or record. Every answer carries
 * a fact_type honesty label and the sources it rests on.
 */
import apiClient from "./client";

type APISuccess<T> = { data: T; success: true };
const base = (farmId: string) => `/farms/${farmId}/bsf/aria`;

export interface AriaAnswer {
  provider: string;
  engine: string;
  fact_type: string; // recorded | calculated | forecast | ai_suggestion | unavailable
  answer: string;
  sources: string[];
  confidence: string;
  ai_enabled: boolean;
}

export async function askAria(farmId: string, question: string): Promise<AriaAnswer> {
  const { data } = await apiClient.post<APISuccess<AriaAnswer>>(`${base(farmId)}/ask`, { question });
  return data.data;
}

export async function getAriaContext(farmId: string): Promise<Record<string, unknown>> {
  const { data } = await apiClient.get<APISuccess<Record<string, unknown>>>(`${base(farmId)}/context`);
  return data.data;
}

File: backend/app/api/v1/endpoints/bsf_aria.py
"""
Greena — BSF ARIA API (Module 16, Part 8)

ARIA explains, summarises, recommends and answers questions about BSF production —
deterministic-first, honesty-labelled, source-referenced. It is read-only: there
is no path by which ARIA mutates a plan or any record (ARIA/Mission Control
Contract). Reuses the platform AI router; `AI_QUERY` guards the question route,
`AI_INSIGHT_VIEW` the read.

Route map (prefix /farms/{farm_id}/bsf/aria):
  POST /ask        ask a question (deterministic-first, grounded LLM fallback)
  GET  /context    the bounded deterministic context ARIA reasons over
"""

from uuid import UUID

from fastapi import APIRouter, Depends

from app.core.permissions import Permission, require_permission
from app.dependencies import CurrentUser, DBSession, require_farm_access
from app.schemas.base import SuccessResponse
from app.schemas.bsf import AriaAnswer, AriaAsk
from app.services import bsf_aria_service as svc

router = APIRouter(prefix="/farms/{farm_id}/bsf/aria", tags=["Black Soldier Fly"])


@router.post("/ask", response_model=SuccessResponse[AriaAnswer])
async def ask(
    farm_id: UUID, body: AriaAsk, db: DBSession, current_user: CurrentUser,
    access: tuple = Depends(require_farm_access()),
    _perm=Depends(require_permission(Permission.AI_QUERY)),
):
    farm, _ = access
    result = await svc.ask(db, farm, current_user, body.question)
    return SuccessResponse(data=AriaAnswer(**result))


@router.get("/context", response_model=SuccessResponse[dict])
async def context(
    farm_id: UUID, db: DBSession, current_user: CurrentUser,
    access: tuple = Depends(require_farm_access()),
    _perm=Depends(require_permission(Permission.AI_INSIGHT_VIEW)),
):
    farm, _ = access
    return SuccessResponse(data=await svc.compile_context(db, farm))
